### PR TITLE
define hammerhead specific window properties as non-enumerable,non-configurable and non-writable (where it is possible) (close #621)

### LINKED ISSUE #621: Code instrumentation window properties should be defined as non-enumerable and non-configurable
Properties like https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/sandbox/code-instrumentation/index.js#L38 should be defined this way https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/index.js#L166

I wonder if there will be any negative performance impact.

### PR DESCRIPTION
**Pull request by miherlosev**
@churkin @AlexanderMoskovkin

### CODE REVIEW

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@inikulin @AlexanderMoskovkin review please

File: src/client/sandbox/code-instrumentation/index.js [hunk 1]
```diff
@@ -35,37 +35,43 @@ export default class CodeInstrumentation extends SandboxBase {
...
-            else if (typeof script === 'string')
-                return processScript(script, false, false);
+                else if (typeof script === 'string')
+                    return processScript(script, false, false);
 
-            return script;
-        };
+                return script;
+            },
+            configurable: true
```
Comment (inikulin) [inline src/client/sandbox/code-instrumentation/index.js]: Why it's configurable?

Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/index.js]: In Google Chrome iframes with html in src raise `load` event twice.
So we define codeinstrumentation function twice.

Comment (inikulin) [inline src/client/sandbox/code-instrumentation/index.js]: We need to add comments which will describe this behavior

**Comment by inikulin** [comment]
\r-

**Comment by miherlosev** [comment]
FPR

**Comment by inikulin** [comment]
Still can't find the comments about purpose in all places there we have configurable properties
\r-

**Comment by miherlosev** [comment]
FPR

File: src/client/sandbox/code-instrumentation/index.js [hunk 2]
```diff
@@ -29,43 +29,51 @@ export default class CodeInstrumentation extends SandboxBase {
     attach (window) {
         super.attach(window);
 
+        // NOTE: In Google Chrome, iframes whose src contains html code raise the 'load' event twice.
+        // So, we need to define code instrumentation functions as 'configurable' so that they can be redefined.
```
Comment (inikulin) [inline src/client/sandbox/code-instrumentation/index.js]: This should be move to the line 39

**Comment by inikulin** [comment]
lgtm with a single style nit

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by miherlosev** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by miherlosev** [comment]
ping @inikulin

File: src/processing/script/header.js [hunk 3]
```diff
@@ -16,14 +16,16 @@ const HEADER = [
...
-    `var ${ INSTRUCTION.getLocation }=__w$?__w$.${ INSTRUCTION.getLocation }:function(l){return l},`,
-    `${ INSTRUCTION.setLocation }=__w$?__w$.${ INSTRUCTION.setLocation }:function(l,v){return l = v},`,
-    `${ INSTRUCTION.setProperty }=__w$?__w$.${ INSTRUCTION.setProperty }:function(o,p,v){return o[p] = v},`,
-    `${ INSTRUCTION.getProperty }=__w$?__w$.${ INSTRUCTION.getProperty }:function(o,p){return o[p]},`,
-    `${ INSTRUCTION.callMethod }=__w$?__w$.${ INSTRUCTION.callMethod }:function(o,p,a){return o[p].apply(o,a)},`,
-    `${ INSTRUCTION.getEval }=__w$?__w$.${ INSTRUCTION.getEval }:function(e){return e},`,
-    `${ INSTRUCTION.processScript }=__w$?__w$.${ INSTRUCTION.processScript }:function(s){return s},`,
-    `${ INSTRUCTION.getStorage }=__w$?__w$.${ INSTRUCTION.getStorage }:function(s){return s};`,
+    'if(!__w$){',
+    `var ${ INSTRUCTION.getLocation }=function(l){return l},`,
```
Comment (inikulin) [inline src/processing/script/header.js]: We can avoid global `__w$` variable now:
```js
if(typeof window!=="undefined"&&window;){ 
 window["${INTERNAL_PROPS.processDomMethodName}"] && window["${INTERNAL_PROPS.processDomMethodName}"]();  
...
} 
```

**Comment by inikulin** [comment]
\r-

**Comment by miherlosev** [comment]
FPR

**Comment by miherlosev** [comment]
ping @inikulin

**Comment by inikulin** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Pull request merged**